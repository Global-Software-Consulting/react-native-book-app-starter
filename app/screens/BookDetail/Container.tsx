import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { NativeEventEmitter, NativeModules, ScrollView, View } from 'react-native';
import { Button, Text } from 'react-native-paper';
import Icon from 'react-native-vector-icons/FontAwesome';
import BookCard from '../../components/BookCard/BookCard';
import Tts from 'react-native-tts';
import { useStyles } from './styles';
import { Props } from './types';
import { useNavigation } from '@react-navigation/core';
import { TouchableOpacity } from 'react-native-gesture-handler';

const Container: React.FC<Props> = (props) => {
    const dummyImages = [
        'https://d1csarkz8obe9u.cloudfront.net/posterpreviews/sad-romance-kindle-book-cover-flyer-template-539b5fdf2dd2f6602c25ce81fbb5d877.jpg?ts=1589326539',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSO4yGwkz128dpVHBztwERbm6Z6kIXwQ03V0A&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQQB-3-cJ6tpQ6cFUjCach0R1dTiXdhXI0AtA&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRuaVjRvjIZAO6XIsd0eHS22Q949ke7r-KZFg&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRkIVlYNBoYdfq1FGWs_dx_k8Irvjmdu64p7A&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRwor6J4qPx1WWN27GYiZEFP-NAoWSstn-YrQ&usqp=CAU',
    ];

    const generateRandomURL = () => {
        const number = Math.random() * 6;
        const index = Math.floor(number);
        return dummyImages[index];
    };
    //theme handling
    const styles = useStyles();
    const navigation = useNavigation();
    const { books } = props;
    const author = 'Dummy author';
    const genre = 'Dummy Genre';
    const pages = books.numberOfPages ?? '111';
    const launched = books.createdAt ?? '11-11-11';
    const title = books.title ?? 'Dummy Title';
    const { t } = useTranslation();
    const [textShown, setTextShown] = useState(false); //To show ur remaining Text
    const [isSpeaking, setIsSpeaking] = useState(false);

    useEffect(() => {
        const ee = new NativeEventEmitter(NativeModules.TextToSpeech);
        ee.addListener('tts-start', () => {});
        ee.addListener('tts-finish', () => {});
        ee.addListener('tts-cancel', () => {});
        if (isSpeaking === true) {
            Tts.speak(books.shortSummary);
        } else {
            Tts.stop();
        }
    }, [isSpeaking]);

    const toggleNumberOfLines = () => {
        //To toggle the show text or hide it
        setTextShown(!textShown);
    };

    return (
        <View style={styles.main}>
            <ScrollView>
                <BookCard
                    id={books?.id ?? 1}
                    styleSelect="ExtraLarge"
                    bookTitle={title}
                    authorName={author}
                    hideIcon={true}
                    url={generateRandomURL()}
                />
                <View
                    style={{
                        flexDirection: 'row',
                        alignItems: 'center',
                        justifyContent: 'center',
                    }}>
                    <View
                        style={[
                            styles.submitView,
                            { backgroundColor: !isSpeaking ? '#00416A' : 'red' },
                        ]}>
                        <Button
                            onPress={() => {
                                setIsSpeaking(!isSpeaking);
                            }}
                            style={styles.submit}>
                            <Text style={{ color: 'white' }}>
                                {!isSpeaking ? t('Speak') : t('Stop')}
                            </Text>
                        </Button>
                    </View>
                    <View style={[styles.submitView, { backgroundColor: '#00416A' }]}>
                        <Button
                            onPress={() => navigation.navigate('BookReader' as never)}
                            style={styles.submit}>
                            <Text style={{ color: 'white' }}>Read Book</Text>
                        </Button>
                    </View>
                </View>
                <View style={styles.horizontalRuler} />

                <View style={styles.mainView}>
                    <View style={styles.subView}>
                        <Text style={styles.genreText}>{t('Genre')}</Text>
                        <Text style={styles.dynamicGenreText}>{genre}</Text>
                    </View>

                    <View style={styles.launchedSubView}>
                        <Text style={styles.launchedText}> {t('Launched')}</Text>
                        <Text style={styles.dynamicLaunchedText}>{launched.substring(0, 9)}</Text>
                    </View>

                    <View style={styles.pagesSubView}>
                        <Text style={styles.sizeText}>{t('Size')}</Text>
                        <Text style={styles.dynamicSize}>
                            {pages}
                            {t(' pages')}
                        </Text>
                    </View>
                </View>

                <View style={styles.synopseView}>
                    <View style={{ flexDirection: 'row' }}>
                        <Text style={styles.synopseText}>{t('Synopse')}</Text>
                    </View>
                    <Text
                        style={styles.dynamicSynopse}
                        onPress={toggleNumberOfLines}
                        numberOfLines={textShown ? undefined : 2}>
                        {books.shortSummary}
                    </Text>
                </View>
            </ScrollView>
        </View>
    );
};

export default Container;
