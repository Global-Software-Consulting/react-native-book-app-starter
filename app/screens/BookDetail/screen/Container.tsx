import React, {useState} from 'react';
import {useTranslation} from 'react-i18next';
import {ScrollView, StyleSheet, View} from 'react-native';
import {Text} from 'react-native-paper';
import {useStyles} from '../styles';
import BookCard from '../../../components/BookCard/BookCard';
import i18n from '../../../config/Languages';
import {heightPercentageToDP} from 'react-native-responsive-screen';

//importing card component
interface Props {
  books?: [];
  base_url?: string;
}
const initI18n = i18n;

const Container: React.FC<Props> = ({books, base_url}) => {
  //theme handling
  const styles = useStyles();
  var author = books.bookAuthors[0].author.name;
  var genre = 'Dummy Genre';
  var pages = books.numberOfPages;
  var launched = books.createdAt;
  var title = books.title;
  const {t, i18n} = useTranslation();

  const [textShown, setTextShown] = useState(false); //To show ur remaining Text
  const [lengthMore, setLengthMore] = useState(false); //to show the "Read more & Less Line"
  const toggleNumberOfLines = () => {
    //To toggle the show text or hide it
    setTextShown(!textShown);
  };
  console.log('Book detailo', books);
  return (
    <View style={styles.main}>
      <ScrollView>
        <BookCard
          styleSelect="ExtraLarge"
          bookTitle={title}
          authorName={author}
          hideIcon={true}
          url="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRoE4lMLbADvLAxUvZf5ZAGvHUZ3KpBWFTW1g&usqp=CAU"
        />

        <View style={styles.horizontalRuler} />

        <View style={styles.mainView}>
          <View style={styles.subView}>
            <Text style={styles.genreText}>{t('Genre')}</Text>
            <Text style={styles.dynamicGenreText}>{genre}</Text>
          </View>

          <View style={styles.launchedSubView}>
            <Text style={styles.launchedText}> {t('Launched')}</Text>
            <Text style={styles.dynamicLaunchedText}>
              {launched.substring(0, 9)}
            </Text>
          </View>

          <View style={styles.pagesSubView}>
            <Text style={styles.sizeText}>{t('Size')}</Text>
            <Text style={styles.dynamicSize}>
              {pages}
              {t(' pages')}
            </Text>
          </View>
        </View>

        <View style={styles.synopseView}>
          <Text style={styles.synopseText}>{t('Synopse')}</Text>
          <Text
            style={styles.dynamicSynopse}
            onPress={toggleNumberOfLines}
            numberOfLines={textShown ? undefined : 2}>
            {books.shortSummary}
          </Text>
        </View>
      </ScrollView>
    </View>
  );
};

export default Container;
