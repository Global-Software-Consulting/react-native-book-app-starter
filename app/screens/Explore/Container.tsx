import { useNavigation } from '@react-navigation/native';
//importing card component
import BookCard from 'components/BookCard/BookCard';
import images from 'config/images';
import { Books, IAppState } from 'models/reducers/appReducers';
import React from 'react';
import { useTranslation } from 'react-i18next';
import {
    FlatList,
    Image,
    RefreshControl,
    ScrollView,
    TouchableHighlight,
    View,
} from 'react-native';
import { Text, useTheme } from 'react-native-paper';
import { useSelector } from 'react-redux';
import { useStyles } from 'screens/Explore/styles';
import { Props } from './types';
const ExploreComponent: React.FC<Props> = (props) => {
    //theme handling
    const styles = useStyles();
    const { t } = useTranslation();
    const theme = useTheme();
    const navigation = useNavigation();
    const { name, onRefresh } = props;
    const dummyImages = [
        'https://d1csarkz8obe9u.cloudfront.net/posterpreviews/sad-romance-kindle-book-cover-flyer-template-539b5fdf2dd2f6602c25ce81fbb5d877.jpg?ts=1589326539',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSO4yGwkz128dpVHBztwERbm6Z6kIXwQ03V0A&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQQB-3-cJ6tpQ6cFUjCach0R1dTiXdhXI0AtA&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRuaVjRvjIZAO6XIsd0eHS22Q949ke7r-KZFg&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRkIVlYNBoYdfq1FGWs_dx_k8Irvjmdu64p7A&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRwor6J4qPx1WWN27GYiZEFP-NAoWSstn-YrQ&usqp=CAU',
    ];

    const generateRandomURL = () => {
        const number = Math.random() * 6;
        const index = Math.floor(number);
        return dummyImages[index];
    };
    //getting data from store
    const books = useSelector((state: { appReducer: IAppState }) => state.appReducer.books);
    const isLoading = useSelector(
        (state: { loadingReducer: IAppState }) => state.loadingReducer.isLoading,
    );

    const navigateToDetails = (params: number) => {
        navigation.navigate('BookDetail' as never, params as never);
    };

    const bookList = (item: Books, style: 'General' | 'Custom') => {
        return (
            <TouchableHighlight
                key={item.id}
                underlayColor={theme.colors.highlight}
                onPress={() => {
                    navigateToDetails(item.id);
                }}>
                <BookCard
                    url={generateRandomURL()}
                    styleSelect={style}
                    bookTitle={item?.title}
                    book={item}
                    id={item?.id}
                />
            </TouchableHighlight>
        );
    };

    return (
        <View style={styles.mainViewSetting}>
            <ScrollView
                nestedScrollEnabled={true}
                style={styles.container}
                showsVerticalScrollIndicator={false}
                refreshControl={
                    <RefreshControl
                        refreshing={isLoading}
                        onRefresh={onRefresh}
                        colors={['green', 'yellow']}
                    />
                }>
                <View style={styles.middleView}>
                    <Text style={styles.name}>
                        {t('Hi')} {name}{' '}
                    </Text>
                    <Text style={styles.tagLine}>{t('Lets find something new')}</Text>

                    {books?.length > 0 ? (
                        <View>
                            <View style={styles.horizontalRuler} />
                            <Text style={styles.listCaption}>{t('Trending')}</Text>
                            <FlatList
                                nestedScrollEnabled={true}
                                horizontal
                                data={books?.filter((item: Books) => {
                                    return item.averageRating > 3;
                                })}
                                contentContainerStyle={styles.flatList}
                                renderItem={({ item }) => bookList(item, 'Custom')}
                                showsHorizontalScrollIndicator={false}
                                keyExtractor={(item, index) => 'key' + index}
                            />
                            <View style={styles.horizontalRuler} />
                            <Text style={styles.listCaption}>{t('New Releases')}</Text>
                            <FlatList
                                horizontal
                                data={books?.filter((item: Books) => {
                                    return item.averageRating <= 3 && item.averageRating > 0;
                                })}
                                contentContainerStyle={styles.flatList}
                                renderItem={({ item }) => bookList(item, 'General')}
                                showsHorizontalScrollIndicator={false}
                                keyExtractor={(item, index) => 'key' + index}
                            />
                            <View style={styles.horizontalRuler} />
                            <Text style={styles.listCaption}>{t('Selected for you')}</Text>
                            <FlatList
                                horizontal
                                data={books?.filter((item: Books) => {
                                    return item.averageRating === 0;
                                })}
                                contentContainerStyle={styles.flatListLast}
                                renderItem={({ item }) => bookList(item, 'General')}
                                showsHorizontalScrollIndicator={false}
                                keyExtractor={(item, index) => 'key' + index}
                            />
                        </View>
                    ) : (
                        <View style={styles.favoriteView}>
                            <Image source={images.books.noBookFound} style={styles.imageError} />
                            <Text style={styles.bookmark}>No books found</Text>
                        </View>
                    )}
                </View>
            </ScrollView>
        </View>
    );
};

export default ExploreComponent;
