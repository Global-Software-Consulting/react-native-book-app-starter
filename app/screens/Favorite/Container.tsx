import { useNavigation } from '@react-navigation/native';
//importing card component
import BookCard from 'components/BookCard/BookCard';
import images from 'config/images';
import { FavoriteBook } from 'models/reducers/appReducers';
import { ReducerState } from 'models/reducers/index';
import React, { useState } from 'react';
import { Dimensions, FlatList, Image, ScrollView, TouchableHighlight, View } from 'react-native';
import { Text, useTheme } from 'react-native-paper';
import { useSelector } from 'react-redux';
import { useStyles } from './styles';
import { Props } from './types';

const Container: React.FC<Props> = (props) => {
    const dummyImages = [
        'https://d1csarkz8obe9u.cloudfront.net/posterpreviews/sad-romance-kindle-book-cover-flyer-template-539b5fdf2dd2f6602c25ce81fbb5d877.jpg?ts=1589326539',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSO4yGwkz128dpVHBztwERbm6Z6kIXwQ03V0A&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQQB-3-cJ6tpQ6cFUjCach0R1dTiXdhXI0AtA&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRuaVjRvjIZAO6XIsd0eHS22Q949ke7r-KZFg&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRkIVlYNBoYdfq1FGWs_dx_k8Irvjmdu64p7A&usqp=CAU',
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRwor6J4qPx1WWN27GYiZEFP-NAoWSstn-YrQ&usqp=CAU',
    ];

    const generateRandomURL = () => {
        const number = Math.random() * 6;
        const index = Math.floor(number);
        return dummyImages[index];
    };
    //theme handling
    const theme = useTheme();
    const styles = useStyles();
    const windowWidth = Dimensions.get('window').width;
    const windowHeight = Dimensions.get('window').height;
    const navigation = useNavigation();
    const isLoading = useSelector((state: ReducerState) => state.loadingReducer.isLoading);
    const favoriteBooks = useSelector((state: ReducerState) => state.appReducer.favorite);
    const { onRefresh } = props;
    const navigateToDetails = async (params: number) => {
        //to check if the internet connection is working
        navigation.navigate('BookDetail' as never, params as never);
    };
    const bookList = (item: FavoriteBook) => {
        return (
            <TouchableHighlight
                key={item.id}
                underlayColor={theme.colors.highlight}
                onPress={() => {
                    navigateToDetails(item.bookId);
                }}>
                <BookCard
                    url={generateRandomURL()}
                    styleSelect="Large"
                    id={item?.book.id}
                    bookTitle={item?.book?.title}
                    book={favoriteBooks}
                    hideIcon={false}
                />
            </TouchableHighlight>
        );
    };
    return (
        <View style={styles.mainView}>
            {favoriteBooks?.length > 0 ? (
                <FlatList
                    numColumns={windowHeight > windowWidth ? 2 : 4}
                    key={windowHeight > windowWidth ? 2 : 4}
                    keyExtractor={(item, index) => 'key' + index}
                    showsVerticalScrollIndicator={false}
                    data={favoriteBooks}
                    style={styles.flatList}
                    contentContainerStyle={{ flexGrow: 1 }}
                    onRefresh={onRefresh}
                    refreshing={isLoading}
                    renderItem={({ item }) => bookList(item)}
                    showsHorizontalScrollIndicator={false}
                />
            ) : (
                <ScrollView contentContainerStyle={styles.favoriteView}>
                    <Image source={images.books.noBookFound} style={styles.imageError} />
                    <Text style={styles.bookmark}>No bookmarks available</Text>
                </ScrollView>
            )}
        </View>
    );
};

export default Container;
