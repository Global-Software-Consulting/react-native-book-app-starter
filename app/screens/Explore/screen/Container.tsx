import React, {useEffect, useState} from 'react';
import {useTranslation} from 'react-i18next';
import {
  Alert,
  FlatList,
  Image,
  RefreshControl,
  ScrollView,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';
import {Text, useTheme} from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {useDispatch, useSelector} from 'react-redux';
import {useStyles} from 'screens/Explore/styles';
//importing card component
import BookCard from 'components/BookCard/BookCard';
import i18n from 'config/Languages/index';
import * as appActions from 'store/actions/appActions';
import * as loginActions from 'store/actions/loginActions';
import {useNavigation} from '@react-navigation/native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  heightPercentageToDP,
  widthPercentageToDP,
} from 'react-native-responsive-screen';
import images from 'config/images';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import Toast from 'react-native-simple-toast';
import NetworkUtils from 'utils/networkUtils';
import {useIsFocused} from '@react-navigation/native';

interface Props {
  books?: [];
  name?: string;
  base_url?: string;
}
const initI18n = i18n;

const ExploreComponent: React.FC<Props> = ({name, base_url}) => {
  //theme handling
  const styles = useStyles();
  const {t, i18n} = useTranslation();
  const theme = useTheme();
  const dispatch = useDispatch();
  const [searchText, setSearchText] = useState('');
  const navigation = useNavigation();
  //getting data from store
  const books = useSelector(state => state.appReducer.detail);
  const isFocused = useIsFocused();
  const isLoading = useSelector(state => state.appReducer.isFetching);
  const favoriteBooks = useSelector(state => state.appReducer.favorite);
  const token = useSelector(state => state.loginReducer.token);
  const newFavorites: string[] = favoriteBooks;
  const searchBook = (bookName: string) => {
    dispatch(appActions.IFetchBooksRequest(bookName));
  };
  const [username, setUserName] = useState(
    userData.firstName + ' ' + userData.lastName,
  );
  const fetchBookDetails = async () => {
    const isConnected = await NetworkUtils.isNetworkAvailable();
    if (!isConnected) {
      dispatch(appActions.IFetchBooksRequest('a'));
      getFavoriteBooks();
    } else {
      Toast.show('You are offline', Toast.SHORT);
    }
  };

  //fetching favorite books
  const getFavoriteBooks = async () => {
    dispatch(appActions.IFetchFavoriteBooksRequest());
  };

  const navigateToDetails = async params => {
    const isConnected = await NetworkUtils.isNetworkAvailable();
    if (!isConnected) {
      navigation.navigate('BookDetail', params);
    } else {
      Toast.show('You are offline', Toast.SHORT);
    }
  };

  return (
    <View style={styles.mainViewSetting}>
      {books.length > 0 ? (
        <ScrollView
          nestedScrollEnabled={true}
          style={styles.container}
          refreshControl={
            <RefreshControl
              refreshing={isLoading}
              onRefresh={fetchBookDetails}
            />
          }>
          <View>
            {/* Searchbar */}
            <View style={styles.searchView}>
              <TextInput
                underlineColorAndroid="transparent"
                placeholder={t('Search Here')}
                placeholderTextColor={theme.colors.text}
                onChangeText={text => setSearchText(text)}
                style={styles.searchViewChildren}
                onEndEditing={() => searchBook(searchText)}
              />
              <Icon
                name="find-in-page"
                size={30}
                style={styles.searchViewChildren}
                onPress={() => searchBook(searchText)}
              />
            </View>
            <Text style={styles.name}>
              {t('Hi')} {name}{' '}
            </Text>
            <Text style={styles.tagLine}>{t('Lets find something new')}</Text>

            <View style={styles.horizontalRuler} />

            <Text style={styles.listCaption}>{t('Trending')}</Text>
            <FlatList
              nestedScrollEnabled={true}
              horizontal
              data={books?.filter(item => {
                return item?.averageRating > 3;
              })}
              contentContainerStyle={styles.flatList}
              renderItem={({item, index}) => (
                <TouchableHighlight
                  key={item}
                  underlayColor="grey"
                  onPress={() => {
                    navigateToDetails(item);
                  }}>
                  <BookCard
                    url={
                      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRoE4lMLbADvLAxUvZf5ZAGvHUZ3KpBWFTW1g&usqp=CAU'
                    }
                    isFavorite={
                      newFavorites.indexOf(item.title) !== -1 ? true : false
                    }
                    styleSelect="Custom"
                    bookTitle={item?.title}
                    book={item}
                    id={item?.id}
                  />
                </TouchableHighlight>
              )}
              showsHorizontalScrollIndicator={false}
            />

            <View style={styles.horizontalRuler} />

            <Text style={styles.listCaption}>{t('New Releases')}</Text>
            <FlatList
              horizontal
              data={books?.filter(item => {
                return item?.averageRating <= 3 && item?.averageRating > 0;
              })}
              contentContainerStyle={styles.flatList}
              renderItem={({item, index}) => (
                <TouchableHighlight
                  key={item}
                  underlayColor="grey"
                  onPress={() => {
                    navigateToDetails(item);
                  }}>
                  <BookCard
                    url={
                      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRoE4lMLbADvLAxUvZf5ZAGvHUZ3KpBWFTW1g&usqp=CAU'
                    }
                    styleSelect="General"
                    bookTitle={item?.title}
                    book={item}
                    id={item?.id}
                  />
                </TouchableHighlight>
              )}
              showsHorizontalScrollIndicator={false}
            />

            <View style={styles.horizontalRuler} />

            <Text style={styles.listCaption}>{t('Selected for you')}</Text>
            <FlatList
              horizontal
              data={books?.filter(item => {
                return item?.averageRating == 0;
              })}
              contentContainerStyle={styles.flatListLast}
              renderItem={({item, index}) => (
                <TouchableHighlight
                  key={item}
                  underlayColor="grey"
                  onPress={() => {
                    navigateToDetails(item);
                  }}>
                  <BookCard
                    url={
                      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRoE4lMLbADvLAxUvZf5ZAGvHUZ3KpBWFTW1g&usqp=CAU'
                    }
                    isFavorite={
                      newFavorites.indexOf(item?.title) !== -1 ? true : false
                    }
                    styleSelect="General"
                    bookTitle={item?.title}
                    book={item}
                    id={item?.id}
                  />
                </TouchableHighlight>
              )}
              showsHorizontalScrollIndicator={false}
            />
          </View>
        </ScrollView>
      ) : (
        <View style={styles.favoriteView}>
          <Image source={images.books.noBookFound} style={styles.imageError} />
          <Text style={styles.bookmark}>
            No books available or check your internet
          </Text>
        </View>
      )}
    </View>
  );
};

export default ExploreComponent;
