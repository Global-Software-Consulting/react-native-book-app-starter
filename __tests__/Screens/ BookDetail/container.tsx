import { useIsFocused } from '@react-navigation/native'
import { render } from '@testing-library/react-native'
import React from 'react'
import * as redux from 'react-redux'

import { act } from 'react-test-renderer'
import Container from 'screens/BookDetail/Container'
import * as utils from 'utils/dimentionUtil'
import { getPercentageHeight, getPercentageWidth } from 'utils/dimentionUtil'
const DeviceTypeUtilsMock = jest.requireMock('utils/dimentionUtil')

jest.mock('react-native-device-info', () => {
    return {
        DeviceInfo: { isTablet: jest.fn() },
        isTablet: jest.fn(),
    }
})
jest.mock('react-redux', () => {
    return {
        __esModule: true,
        A: true,
        useDispatch: jest.fn(),
        useSelector: jest.fn(),
        default: 'mockedDefaultExport',
    }
})
jest.mock('react-native-paper', () => {
    return {
        Text: jest.fn(),
        Button: jest.fn(),
        useTheme: jest.fn(),
    }
})
jest.mock('./../../../app/utils/dimentionUtil', () => {
    return {
        getPercentageHeight: jest.fn(),
        height: '100',
        getPercentageWidth: jest.fn(),
    }
})

// jest.mock('react-native', () => {
//     return {
//         NativeEventEmitter: jest.fn(),
//         NativeModules: jest.fn(),
//         StyleSheet: {
//             create: () => ({}),
//         },
//         Dimensions: {
//             get: () => ({
//                 width: jest.fn(),
//                 height: jest.fn(),
//             }),
//         },
//     }
// })

jest.mock('@react-navigation/core', () => {
    return {
        useIsFocused: jest.fn(),
        useNavigation: jest.fn(),
    }
})
jest.mock('react-native-image-crop-picker', () => {
    return {
        __esModule: true,
        A: true,
        useDispatch: jest.fn(),
        default: 'mockedDefaultExport',
    }
})

jest.mock('react-native-keyboard-aware-scroll-view', () => {
    return {
        __esModule: true,
        A: true,
        KeyboardAwareScrollView: jest.fn(),
        default: 'mockedDefaultExport',
    }
})
jest.mock('react-native-linear-gradient', () => {
    return {
        __esModule: true,
        A: true,
        LinearGradient: jest.fn(),
        default: 'mockedDefaultExport',
    }
})
jest.mock('./../../../app/config/images', () => {
    return {
        __esModule: true,
        A: true,
        app: { logo: jest.fn() },
        default: 'mockedDefaultExport',
    }
})

jest.mock('react-native-tts', () => {
    return {
        __esModule: true,
        A: true,
        Tts: jest.fn(),
        default: 'mockedDefaultExport',
    }
})

describe('Component testing', () => {
    test('Snapshot ', () => {
        const height = jest.fn()
        getPercentageHeight.mockReturnValue(jest.fn())

        const width = jest.fn()
        getPercentageWidth.mockReturnValue(jest.fn())
        const tree = render(<Container books={['abc']} base_url="abc.com" />)
        expect(tree).toMatchSnapshot()
    })
})
