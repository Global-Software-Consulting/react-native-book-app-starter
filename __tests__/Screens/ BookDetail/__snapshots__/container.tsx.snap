// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Snapshot  1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "minHeight": undefined,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "elevation": 2,
            "margin": 2,
            "marginRight": 3,
            "shadowColor": "white",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 3,
            "shadowRadius": 2,
          }
        }
      >
        <mockedDefaultExport
          resizeMode="contain"
          source={
            Object {
              "priority": undefined,
              "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRkIVlYNBoYdfq1FGWs_dx_k8Irvjmdu64p7A&usqp=CAU",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "borderRadius": 20,
              "height": undefined,
              "margin": 5,
              "width": undefined,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "alignSelf": "center",
                "fontSize": undefined,
                "fontWeight": "500",
                "maxWidth": undefined,
                "textAlign": "center",
              },
            ]
          }
        >
          Dummy Title
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "alignSelf": "center",
                "color": "red",
                "fontSize": undefined,
                "fontWeight": "300",
                "maxWidth": undefined,
              },
            ]
          }
        >
          Dummy author
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#00416A",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "margin": 2,
              "marginTop": 30,
              "width": undefined,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="play"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                }
              }
            >
              Pause
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#00416A",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "margin": 2,
              "marginTop": 30,
              "width": undefined,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="navigateToRead"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                }
              }
            >
              Read Book
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "backgroundColor": "#00416A",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "margin": 2,
              "marginTop": 30,
              "width": undefined,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="navigate"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                }
              }
            >
              Share
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#DCDCDC",
            "borderBottomWidth": 1,
            "marginTop": 4,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "300",
                "textAlign": "center",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "maxWidth": undefined,
                "textAlign": "center",
              }
            }
          >
            Dummy Genre
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "300",
                "textAlign": "center",
              }
            }
          >
             
          </Text>
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "maxWidth": undefined,
                "textAlign": "center",
              }
            }
          >
            11-11-11
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "300",
                "textAlign": "center",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "maxWidth": undefined,
                "textAlign": "center",
              }
            }
          >
            111
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "margin": 30,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "300",
                "marginBottom": 10,
                "marginRight": 10,
              }
            }
          />
        </View>
        <Text
          numberOfLines={2}
          onPress={[Function]}
          style={
            Object {
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 50,
            }
          }
          testID="toggle"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
